src/Components/Trainers/Form/index.jsx

```diff
@@ -3,88 +3,60 @@ import { useHistory, useParams } from 'react-router-dom';
 import styles from './form.module.css';
 import Button from '../../Shared/Button/index';
 import SharedModal from '../../Shared/Modal/index';
+import { useSelector, useDispatch } from 'react-redux';
+import { updateTrainer, addTrainer } from '../../../Redux/trainers/thunks';
 
 const Form = () => {
   const history = useHistory();
   const { id } = useParams();
-  const [firstName, setName] = useState('');
-  const [lastName, setLastName] = useState('');
-  const [dni, setDNI] = useState('');
-  const [phone, setPhoneNumber] = useState('');
-  const [email, setEmail] = useState('');
-  const [city, setCity] = useState('');
-  const [password, setPassword] = useState('');
-  const [salary, setSalary] = useState('');
   const [showModal, setShowModal] = useState(false);
   const [modalMessage, setModalMessage] = useState('');
   const [modalTypeStyle, setModalTypeStyle] = useState('success');
   const [shouldRedirect, setShouldRedirect] = useState(false);
-
-  const getTrainerById = async () => {
-    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers/${id}`);
-    const data = await response.json();
-    const trainer = data.data;
-    setName(trainer.firstName);
-    setLastName(trainer.lastName);
-    setDNI(trainer.dni);
-    setPhoneNumber(trainer.phone);
-    setEmail(trainer.email);
-    setCity(trainer.city);
-    setSalary(trainer.salary);
-    setPassword(trainer.password);
-  };
+  const dispatch = useDispatch();
+  const trainers = useSelector((state) => state.trainers.data);
+  const [trainer, setTrainer] = useState({
+    firstName: '',
+    lastName: '',
+    dni: '',
+    phone: '',
+    email: '',
+    city: '',
+    salary: '',
+    password: ''
+  });
 
   useEffect(() => {
     if (id) {
       getTrainerById();
     }
-  }, []);
-
-  const handleCloseModal = () => {
-    setShowModal(false);
-    if (shouldRedirect) {
-      history.push('/trainers');
-    }
-  };
+  }, [id]);
```

Comment:
I think you can dismiss this Id param inside the useEffect function

Reply:
I prefer to eliminate it from the fas of the earth